src/components/ContactDetails.vue

```diff
@@ -806,14 +806,16 @@ export default {
 		/**
 		 * Generate a qrcode for the contact
 		 */
-		showQRcode() {
+		async showQRcode() {
 			const jCal = this.contact.jCal.slice(0)
 			// do not encode photo
 			jCal[1] = jCal[1].filter(props => props[0] !== 'photo')
 
 			const data = stringify(jCal)
 			if (data.length > 0) {
-				this.qrcode = btoa(qr.imageSync(data, { type: 'svg' }))
+				const svgBytes = await getSVG(data)
+				const svgString = new TextDecoder().decode(svgBytes)
+				this.qrcode = btoa(svgString)
```

Comment:
does the async variant improve bundle size, e.g. because it enables lazy loading?

Reply:
I's just that the old qr package has been unmaintained for a couple of years and I had to switch to a fork that modernized the code a lot.

IIRC, the old one uses node internals which break when bundled with vite while the new one dropped all node internals so that it can be used in a browser without polyfills.

Reply:
Got it